src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+                {
+                    "ingestDelay":  int(x.get("metadata",{}).get("ingestDelay","60s")[:-1]),
+                    "samplePeriod": int(x.get("metadata",{}).get("samplePeriod","60s")[:-1]),
+                    "valueType":    x.get("valueType"),
+                    "metricKind":   x.get("metricKind"),
+                    "unit":         unit_caster(x.get("unit"))
+            },
+            "dimensions": [ 
+                {
+                "key":      dimension.get("key"), 
+                "value":    "label:metric.labels." + dimension.get("key"),
+                } for dimension in x.get("labels") or []],
+            "monitored_resources_types": x.get("monitoredResourceTypes",[])
+        } for x in page["metricDescriptors"]]
+    
+    
+
+
+
+    metric_list = [(Metric(**x), x.get("monitored_resources_types"),x) for x in metric_yaml if x.get("gcpOptions").get("valueType").upper() != "STRING" ]
+    discovered_resource_type_list = list(filter(lambda x: discovered_resource_type in x[1], metric_list))
+    return discovered_resource_type_list
+
+
+async def enrich_services_autodiscovery(gcp_services_list,gcp_session,dt_session,token):
+    print("Start AutoDiscovery")
+
+
+    bucket_gcp_services =  list(filter(lambda x: discovered_resource_type in x.name, gcp_services_list[0]))
+    existing_metric_list = [metric for service in bucket_gcp_services for metric in service.metrics]
+    all_metrics_list = await get_metric_descriptors(gcp_session,token)
+
+
+    missing_metrics_list = [metric for metric in all_metrics_list if metric[0].google_metric not in [existing_metric.google_metric for existing_metric in existing_metric_list]]
+
+
+    print(f"In Extension we have this amount of metrics: {len(existing_metric_list)}" )
+    print(f"Resource type: {discovered_resource_type} have this amount of metrics: {len(all_metrics_list)}" )
+    print(f"Adding {len(missing_metrics_list)} metrics")
+    print(f"Adding metrics: [{[metric[0].google_metric for metric in missing_metrics_list]}]")
+
+    for service in gcp_services_list[0]:
+            if service.name == discovered_resource_type:
+                service.metrics.extend([metric[0] for metric in missing_metrics_list])
+    
+
+    print(f"Adding Metadata to metrics")
+    dynatrace_url = await fetch_dynatrace_url(gcp_session, config.project_id(), token),
+    dynatrace_access_key=await fetch_dynatrace_api_key(gcp_session, config.project_id(), token),
+
+    metric_metadata_list = [MetricMetadata(**(x[2])) for x in missing_metrics_list]
+    #Metadata Metrics
+    #ret = await _sent_metadata_to_dt(dynatrace_url[0],dynatrace_access_key,dt_session,metric_metadata_list)
+    
+    return gcp_services_list
+
+
+
+
+def main():
+    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = r"C:\Users\szymon.nagel\Downloads\dynatrace-gcp-extension-0618ff31e584.json"
+    logging_context.log("Dynatrace GCP Metrics autodiscovery startup")
+    asyncio.run(init_session())
```

Comment:
Why are you running new loop in here???

Reply:
Ahhh, you were trying to run the file yourself in dev approach, right?
Thats fine! Maybe a unit tests for this would be better, and actually can stay as useful unit test for future

Reply:
Btw, before you merge it, please add unit tests